src/Factories/AllEntriesControllerFactory.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace App\Factories;
+
+use App\Controllers\AllEntriesController;
+use Psr\Container\ContainerInterface;
+
+class AllEntriesControllerFactory
+{
+    public function __invoke(ContainerInterface $container)
+    {
+        $BlogModel = $container->get('BlogModel');
+        $renderer = $container->get('renderer');
+        $BlogModelPageController = new AllEntriesController($BlogModel, $renderer);

```

Comment:
you should type hint in the controllers themselves to only allow instances of $BlogModel